megamek/src/megamek/common/Dropship.java

```diff
@@ -270,6 +259,20 @@ public boolean isLocationProhibited(Coords c, int currElevation) {
 
         return isProhibited;
     }
+    
+    /**
+     * Worker function that checks if a given hex contains terrain onto which a grounded dropship
+     * cannot deploy. 
+     */
+    private boolean hexContainsProhibitedTerrain(IHex hex) {
+        return hex.containsTerrain(Terrains.WOODS) || hex.containsTerrain(Terrains.ROUGH)
+                || ((hex.terrainLevel(Terrains.WATER) > 0) && !hex.containsTerrain(Terrains.ICE))
+                || hex.containsTerrain(Terrains.RUBBLE) || hex.containsTerrain(Terrains.MAGMA)
+                || hex.containsTerrain(Terrains.JUNGLE) || (hex.terrainLevel(Terrains.SNOW) > 1)
+                || (hex.terrainLevel(Terrains.GEYSER) == 2) 
+                || hex.containsTerrain(Terrains.BUILDING) || hex.containsTerrain(Terrains.IMPASSABLE);
```

Comment:
Does this also catch bridges? I can't remember.

Reply:
Water implicitly catches bridges over water, but bridges can apparently go over land too, so I better add it in.